src/c2s/mongoose_c2s_ranch.erl

```diff
@@ -78,33 +83,29 @@ socket_handle_data(#state{transport = fast_tls, socket = TlsSocket}, {tcp, _Sock
         {error, Reason} ->
             {error, Reason}
     end;
-socket_handle_data(#state{transport = ranch_ssl, socket = Socket}, {ssl, Socket, Data}) ->
+socket_handle_data(#state{transport = Transport, socket = Socket}, {ssl, Socket, Data})
+      when Transport ==  just_tls; Transport == ranch_ssl ->
     mongoose_metrics:update(global, [data, xmpp, received, encrypted_size], iolist_size(Data)),
     Data;
 socket_handle_data(#state{transport = ranch_tcp, socket = Socket}, {tcp, Socket, Data}) ->
     Data.
 
 -spec socket_activate(state()) -> ok.
-socket_activate(#state{transport = fast_tls, socket = Socket}) ->
-    fast_tls:setopts(Socket, [{active, once}]);
-socket_activate(#state{transport = ranch_ssl, socket = Socket}) ->
-    ranch_ssl:setopts(Socket, [{active, once}]);
-socket_activate(#state{transport = ranch_tcp, socket = Socket}) ->
-    ranch_tcp:setopts(Socket, [{active, once}]).
+socket_activate(#state{transport = Transport, socket = Socket}) ->
+    Mod = mod(Transport),
+    Mod:setopts(Socket, [{active, once}]).
```

Comment:
Here I had thought of being explicit about the three modules because a pattern-match on the module name (an atom) and a fully qualified function call (`module:function(args)`) is theoretically faster than a module name as a variable (in the BEAM, the module name as a variable means going to a big hash-table with all loaded code and looking up what to execute, while a fully qualified function call already contains the function pointer to jump to).

But arguably, the module as a variable is a more common design pattern. I fought against it thinking that only three values doesn't make it all too unreadable. And maybe allows dialyzer to be more precise 😅